prov/verbs/src/verbs_eq.c

```diff
@@ -888,35 +968,46 @@ fi_ibv_eq_read(struct fid_eq *eq_fid, uint32_t *event,
 		return ret;
 	}
 
-	return -FI_EAGAIN;
+	ret = epoll_wait(eq->epfd, &events, 1, timeout);
+	if (ret == 0)
+		return -FI_EAGAIN;
+	else if (ret < 0)
+		return -errno;
+
+	assert(ret > 0);
+	if (events.data.ptr) {
+		if (!domain) {
+			/* Update the domain before retrying if it's not yet set */
+			domain = events.data.ptr;
+		} else
+			assert(domain == events.data.ptr);
+	}
+	goto retry;
```

Comment:
This looks like a while (1) loop.  Please convert.

Reply:
Thanks, I converted the loop into a while(1) loop.